src/Confluent.Kafka/Impl/SafeHandleZeroIsInvalid.cs

```diff
@@ -24,12 +24,31 @@ namespace Confluent.Kafka.Impl
 {
     abstract class SafeHandleZeroIsInvalid : SafeHandle
     {
-        internal SafeHandleZeroIsInvalid() : base(IntPtr.Zero, true) { }
+        private string handleName;
 
-        internal SafeHandleZeroIsInvalid(bool ownsHandle) : base(IntPtr.Zero, ownsHandle) { }
+        internal SafeHandleZeroIsInvalid(string handleName) : base(IntPtr.Zero, true)
+        {
+            this.handleName = handleName;
+        }
 
-        public override bool IsInvalid => handle == IntPtr.Zero;
+        internal SafeHandleZeroIsInvalid(string handleName, bool ownsHandle) : base(IntPtr.Zero, ownsHandle)
+        {
+            this.handleName = handleName;
+        }
 
-        protected override bool ReleaseHandle() => true;
+        public override bool IsInvalid => handle == IntPtr.Zero;
+        
+        /// <summary>
+        /// Prevent AccessViolationException when handle has already been closed
```

Comment:
Indent comments 4 spaces to make consist with other comments. Full stop after 'closed' because in the generated docs, the line break will disappear.